flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -288,10 +288,24 @@ private void configureStreamGraph(final StreamGraph graph) {
 		} else {
 			graph.setStateBackend(stateBackend);
 			graph.setGlobalDataExchangeMode(GlobalDataExchangeMode.ALL_EDGES_PIPELINED);
+			graph.setAllVerticesInSameSlotSharingGroupByDefault(true);
```

Comment:
Why is this change necessary?

Reply:
It seems coming from a code merge (for sorted merge shuffle), but for some reason, it was removed in a later version.

Thanks for catching this. I will remove it.